lumigator/python/mzai/backend/backend/services/experiments.py

```diff
@@ -33,6 +32,9 @@ def __init__(
         self._job_service = job_service
         self._dataset_service = dataset_service
 
+    def _raise_not_found(self, job_id: UUID):
+        raise HTTPException(status.HTTP_404_NOT_FOUND, f"Job {job_id} not found.")
+
```

Comment:
```suggestion
```

We don't need this now, instead just raise a `JobNotFoundError` or `ExperimentNotFoundError`. 😄